test/system/055-rm.bats

```diff
@@ -113,4 +113,21 @@ load helpers
     is "$output" "" "Should print no output"
 }
 
+@test "podman container rm doesn't affect stopping containers" {
+    rand=$(random_string 30)
+    run_podman run --name $rand -d --health-cmd /bin/false --health-interval 1s --health-retries 2 --health-timeout 1s --health-on-failure=stop --stop-timeout=2 --health-start-period 0 --stop-signal SIGTERM $IMAGE sleep infinity
+
+    pid=`podman inspect $rand | jq ".[0].State.Pid"`
```

Comment:
EEEEK! You can't do this. More later, but placing an immediate block right now.

Reply:
Suggestion:
```bash
@test "podman container rm doesn't affect stopping containers" {
    local cname=c$(random_string 30)
    run_podman run -d --name $cname \
               --health-cmd /bin/false \
               --health-interval 1s \
               --health-retries 2 \
               --health-timeout 1s \
               --health-on-failure=stop \
               --stop-timeout=2 \
               --health-start-period 0 \
               --stop-signal SIGTERM \
               $IMAGE sleep infinity
    local cid=$output

    # We'll use the PID later to confirm that container is not running
    run_podman inspect --format '{{.State.Pid}}' $cname
    local pid=$output

    # rm without -f should fail, because container is running/stopping.
    # We have no way to guarantee that we see 'stopping', but at a very
    # minimum we need to check at least one rm failure
    local rm_failures=0
    for i in {1..10}; do
        run_podman '?' rm $cname
        if [[ $status -eq 0 ]]; then
            break
        fi

        # rm failed. Confirm that it's for the right reason.
        assert "$output" =~ "Error: cannot remove container $cid as it is .* - running or paused containers cannot be removed without force: container state improper" \
               "Expected error message from podman rm"
        rm_failures=$((rm_failures + 1))
        sleep 1
    done

    # At this point, container should be gone
    run_podman 1 container exists $cname
    run_podman 1 container exists $cid

    assert "$rm_failures" -gt 0 "we want at least one failure from podman-rm"

    if kill -0 $pid; then
        die "Container $cname process is still running (pid $pid)"
    fi
}
```
Explanation:
* you can't just do `foo=$(podman ...)`. That runs literal `podman`, from `$PATH`, instead of the desired podman-under-test which could be `/path/to/podman`, or `/path/to/podman-remote`, or `/path/to/podman --root-or-something`.
* This adds an actual check to confirm that there was at least one `rm` failure, and for the expected reason. (We can't check for "stopping" because that would be racy, but this is better than nothing)
* Add a few more sanity checks. Clean up the `kill` check, remove an unnecessary `run`. Format the `podman run` line for legibility. Use a better variable name than `rand`. Add comments explaining the not-immediately-obvious stuff.

HTH

Reply:
PID recycling or any other reasons?  And just noticed I should use `$PODMAN` instead of `podman`

Reply:
No, don't use `$PODMAN`. Use `run_podman`.

Reply:
I don't understand your question about PID recycling, but no, that has nothing to do with all the concerns I listed above.

Reply:
thanks, I amended your version and pushed it.

The PID recycling issue is that when we check `kill -0 $pid`, the PID could be already re-used by another process, so a more accurate check would be to verify that it is effectively the same process looking at its starttime.  Although, this is unlikely to happen and will just make the test case more difficult to follow